arci-urdf-viz/src/client.rs

```diff
@@ -206,60 +217,92 @@ impl UrdfVizWebClient {
         })))
     }
 
-    pub fn run_thread(&self) {
+    pub fn run_send_velocity_thread(&self) {
+        if mem::replace(
+            &mut self.0.threads.lock().unwrap().has_send_velocity_thread,
+            true,
+        ) {
+            panic!("send_velocity_thread is running");
+        }
+
         let inner = self.0.clone();
         std::thread::spawn(move || {
-            while !inner.is_dropping.load(Ordering::Relaxed) {
+            struct Bomb(Arc<UrdfVizWebClientInner>);
+            impl Drop for Bomb {
+                fn drop(&mut self) {
+                    self.0.threads.lock().unwrap().has_send_velocity_thread = false;
+                    debug!("terminating send_velocity_thread");
+                }
+            }
+
+            let bomb = Bomb(inner);
+            while !bomb.0.is_dropping() {
                 const DT: f64 = 0.02;
                 let _ = ScopedSleep::from_secs(DT);
-                let velocity = inner.velocity.lock().unwrap();
-                let mut pose = get_robot_origin(&inner.base_url).unwrap();
+                let velocity = bomb.0.velocity.lock().unwrap();
+                let mut pose = get_robot_origin(&bomb.0.base_url).unwrap();
                 let mut rpy = euler_angles_from_quaternion(&pose.quaternion);
                 let yaw = rpy.2;
                 pose.position[0] += (velocity.x * yaw.cos() - velocity.y * yaw.sin()) * DT;
                 pose.position[1] += (velocity.x * yaw.sin() + velocity.y * yaw.cos()) * DT;
                 rpy.2 += velocity.theta * DT;
                 pose.quaternion = quaternion_from_euler_angles(rpy.0, rpy.1, rpy.2);
-                send_robot_origin(&inner.base_url, pose).unwrap();
+                send_robot_origin(&bomb.0.base_url, pose).unwrap();
             }
         });
     }
 
-    pub fn run_send_joint_positions_thread(&mut self) {
-        if self
-            .0
-            .has_send_joint_positions_thread
-            .swap(true, Ordering::Relaxed)
-        {
-            panic!("send_joint_positions_thread is running.");
+    pub fn run_send_joint_positions_thread(&self) {
+        if mem::replace(
+            &mut self
+                .0
+                .threads
+                .lock()
+                .unwrap()
+                .has_send_joint_positions_thread,
+            true,
+        ) {
+            panic!("send_joint_positions_thread is running");
         }
 
         let inner = self.0.clone();
         std::thread::spawn(move || {
-            while !inner.is_dropping.load(Ordering::Relaxed) {
+            struct Bomb(Arc<UrdfVizWebClientInner>);
+            impl Drop for Bomb {
+                fn drop(&mut self) {
+                    self.0
+                        .threads
+                        .lock()
+                        .unwrap()
+                        .has_send_joint_positions_thread = false;
+                    debug!("terminating send_joint_positions_thread");
+                }
+            }
```

Comment:
Previously, if a panic occurred in the thread, the thread would terminate, but panic would not be propagated to the main thread, and subsequent `send_joint_positions` calls would never be sent to urdf-viz.

This patch sets the flag to false when the thread terminated so that subsequent `send_joint_positions` calls are panicked if the termination of the thread is due to panic.